parameter/DomainConfiguration.cpp

```diff
@@ -50,11 +53,8 @@ CDomainConfiguration::CDomainConfiguration(const string& strName) : base(strName
 
 CDomainConfiguration::~CDomainConfiguration()
 {
-    AreaConfigurationListIterator it;
-
-    for (it = _areaConfigurationList.begin(); it != _areaConfigurationList.end(); ++it) {
-
-        delete *it;
+    for (auto areaConfiguration : _areaConfigurationList) {
+        delete areaConfiguration;
```

Comment:
Use an `std::list<std::unique_ptr<CAreaConfiguration>>` instead: you won't have to implement anything in the destructor.